FailedMessage/FailedMessageRejecter.php

```diff
@@ -0,0 +1,42 @@
+<?php declare(strict_types=1);
+
+namespace KaroIO\MessengerMonitorBundle\FailedMessage;
+
+use KaroIO\MessengerMonitorBundle\Exception\FailureReceiverDoesNotExistException;
+use KaroIO\MessengerMonitorBundle\Exception\FailureTransportNotListable;
+use KaroIO\MessengerMonitorBundle\Exception\FailureReceiverNotListableException;
+use KaroIO\MessengerMonitorBundle\Locator\ReceiverLocator;
+use Symfony\Component\Messenger\Transport\Receiver\ListableReceiverInterface;
+
+class FailedMessageRejecter
+{
+    private $receiverLocator;
+    private $failureReceiverName;
+
+    public function __construct(ReceiverLocator $receiverLocator, ?string $failureReceiverName)
+    {
+        $this->receiverLocator = $receiverLocator;
+        $this->failureReceiverName = $failureReceiverName;
+    }
+
+    public function rejectFailedMessage($id): void
+    {
+        if (null === $this->failureReceiverName) {
+            throw new FailureReceiverDoesNotExistException();
+        }
+
+        $failureReceiver = $this->receiverLocator->getReceiver($this->failureReceiverName);
+
+        if (!$failureReceiver instanceof ListableReceiverInterface) {
+            throw new FailureReceiverNotListableException();
+        }
```

Comment:
This piece of code is repeated three times in the code now.

Any idea to wrap this with an external class?

Reply:
i've created a `FailureReceiverProvider` which also may be usefull in the future when the PR with multiple failure transport will be merged